uber/models/__init__.py

```diff
@@ -557,6 +567,52 @@ def initialize_db(cls, modify_tables=False, drop=False, initialize=False):
                 from uber.migration import stamp
                 stamp('heads' if modify_tables else None)
 
+
+    def __init__(self, force_new_session=False, use_nested_transaction=False):
+        if not force_new_session and threadlocal.get(self._SESSION_KEY):
+            self.session = threadlocal.get(self._SESSION_KEY)
+            self.is_new_session = False
+        else:
+            self.session = self.session_factory()
+            self.is_new_session = True
+
+            import types
+            for name, val in self.SessionMixin.__dict__.items():
+                if not name.startswith('__'):
+                    assert not hasattr(self.session, name) and hasattr(val, '__call__')
+                    setattr(self.session, name, types.MethodType(val, self.session))
+
+        self.use_nested_transaction = use_nested_transaction
```

Comment:
Nice, I didn't know about SAVEPOINT functionality, but this is a great way to expose it in this context manager.